src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/registries/resourceViewRegistry.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+import type {ResourceViewsMap} from '../types';
+
+class ResourceViewRegistry {
+    resourceViews: ResourceViewsMap;
+
+    constructor() {
+        this.clear();
+    }
+
+    clear() {
+        this.resourceViews = {};
+    }
+
+    addResourceViews(resourceViews: ResourceViewsMap) {
+        Object.keys(resourceViews).forEach((resourceKey) => {
+            if (this.resourceViews[resourceKey]) {
+                throw new Error('The resource views for "' + resourceKey + '" has already be configured.');
+            }
+
+            this.resourceViews[resourceKey] = resourceViews[resourceKey];
+        });
+    }
+
+    has(view: string, resourceKey: string): boolean {
+        if (!this.resourceViews[resourceKey]) {
+            return false;
+        }
+
+        return this.resourceViews[resourceKey].views[view] !== undefined;
+    }
+
+    get(view: string, resourceKey: string): string {
+        if (typeof this.resourceViews[resourceKey] !== 'object') {
+            throw new Error('The resource "' + resourceKey + '" was not found.');
+        }
+
+        if (typeof this.resourceViews[resourceKey].views[view] !== 'string') {
+            throw new Error('The resource view "' + view + '" for resource "' + resourceKey + '" was not found.');
```

Comment:
Do we really have to distinguish in the error between `'The resource "' + resourceKey + '" was not found.` and `'The resource view "' + view + '" for resource "' + resourceKey + '" was not found.'` ?

I think the second one should be sufficient and then we could here also use the `has` method

Reply:
Think if the error happens it is good to know if the `resourceKey` was not provided or if the `view` is just missing, think for better DX it make sense to have different specific error messages here.